lms/djangoapps/enthought/auth.py

```diff
@@ -0,0 +1,169 @@
+""" Authenticate the user using Enthought credentials. """
+
+# Standard library.
+import os
+import requests
+
+# 3rd party library.
+from django import forms
+from django.conf import settings
+from django.contrib.auth.models import User
+
+# edX library.
+from student.forms import AccountCreationForm
+
+
+class EnthoughtAuthBackend(object):
+    """ A Django authentication backend that authenticates users via the
+    Enthought web API.
+
+    """
+
+    def authenticate(self, username=None, password=None, request=None):
+        """
+        Try to authenticate a user. This method will try to authenticate the
+        user on the Enthought API system and if successful, return a User
+        object from the local database.
+
+        If a User object is not found in the local database, it creates one.
+        """
+
+        if not username:
+            username = request.POST.get('email')
+
+        if not password:
+            password = request.POST.get('password')
+
```

Comment:
Further, `request` could also be `None`, but is it assumed it is present without checking.